reframe/core/config.py

```diff
@@ -67,6 +66,11 @@ def __init__(self, site_config, filename):
         self._subconfigs = {}
         self._local_system = None
         self._sticky_options = {}
+        self.autodetect_opts = {
+            'autodetect_method': 'hostname',
+            'autodetect_xthostname': True,
+            'autodetect_fqdn': True,
```

Comment:
I don't like very much that we set defaults here as well as in the cli. I would vote to have defaults only in the cli.